src/CMakeLists.txt

```diff
@@ -3,114 +3,183 @@ cmake_policy(SET CMP0054 OLD) # behaviour keywords in CMake if tests
 # ENDIF(COMMAND cmake_policy)
 
 file( GLOB_RECURSE
-      CCTag_hpp
-      ./cctag/*.hpp
-)
-
-file( GLOB_RECURSE
       CCTag_cpp
       ./cctag/*.cpp
 )
 
-if(USE_CUDA)
-    file( GLOB_RECURSE
-          CUDA_cpp
-          ./cuda/*.cpp
-    )
-    file( GLOB_RECURSE
-          CUDA_cu
-          ./cuda/*.cu
-    )
-
-    include_directories("${CMAKE_SOURCE_DIR}/src")
-    include_directories("${CMAKE_SOURCE_DIR}/3rdparty/Cuda-7.0-cub")
-
-    set_source_files_properties( ${CUDA_cu} PROPERTIES COMPILE_FLAGS "-Wall --std=c++11")
-    set_source_files_properties( ${CUDA_cpp} PROPERTIES COMPILE_FLAGS "-Wall --std=c++11")
-    set_source_files_properties( ${CCTag_cpp} PROPERTIES COMPILE_FLAGS "-std=c++11")
-
-    add_definitions("-DSM_ARCH=100")
-    # add_definitions("-DSM_TARGETS=-gencode=arch=compute_52,code=\"sm_52,compute_52\"")
-    add_definitions("-DSM=100")
-
-    # set(CUDA_SEPARABLE_COMPILATION ON)
-    # set(CUDA_ATTACH_VS_BUILD_RULE_TO_CUDA_FILE ON)
-    # set(BUILD_SHARED_LIBS ON)
-
-
-    if(NOT COMPILE_STATIC_CCTAG_LIBRARY)
-    	message( FATAL_ERROR "We must link CCTag library static with CUDA" )
-    endif()
+if(WITH_CUDA)
+  file( GLOB_RECURSE
+        CUDA_cpp
+        ./cuda/*.cpp
+  )
+  file( GLOB_RECURSE
+        CUDA_cu
+        ./cuda/*.cu
+  )
+
+  # add_definitions("-DSM_TARGETS=-gencode=arch=compute_52,code=\"sm_52,compute_52\"")
+
+  # set(CUDA_SEPARABLE_COMPILATION ON)
+  # set(CUDA_ATTACH_VS_BUILD_RULE_TO_CUDA_FILE ON)
+  # set(BUILD_SHARED_LIBS ON)
+
+  if(NOT COMPILE_STATIC_CCTAG_LIBRARY)
+    message( FATAL_ERROR "We must link CCTag library static with CUDA" )
```

Comment:
To use CUDA, CCTag library needs to be build in static.
To disable CUDA and generate a dynamic library use: `-DWITH_CUDA=false`.